### ISSUE TITLE
dgl.random.seed API seems not working for sampling & random walks across different runs

### ISSUE THREAD

**Issue by drjiadu** [dmlc/dgl#1692]
## 🐛 Bug

It seems dgl.random.seed API does not working for sampling & random walks across different runs. 
Within the same run, it works. However, it does not work across runs. Maybe something related to system clock in C module.

## To Reproduce

1. save the code below as a python file (test.py), and run (python test.py)
2. rerun (python test.py) after around 10 seconds, and it will yield different results

```
import dgl
import scipy
import torch
import numpy as np
np.random.seed(1)
dgl.random.seed(1)
torch.manual_seed(1)
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False

g = scipy.sparse.random(3000, 5000, 0.003)
#print(g)

G = dgl.heterograph({
    ('A', 'AB', 'B'): g,
    ('B', 'BA', 'A'): g.T})
seeds = torch.LongTensor([0, 1, 2])

sampler = dgl.sampling.PinSAGESampler(G, 'A', 'B', 3, 0.5, 200, 10)
frontier = sampler(seeds)
print(frontier.all_edges(form='uv'))
```

## Expected behavior

The resulting node index between two runs are different despite of the random seed.

## Environment

 - DGL Version: 0.4.3.post2
 - Backend Library & Version:  PyTorch 1.2.0:
 - OS: Ubuntu 16.0.4
 - How you installed DGL: pip
 - Python version: python 3.5
 - Any other relevant information:

**Comment by BarclayII** [comment]
We use OpenMP to parallelize random walks and that may be why the results differ.  Could you set OpenMP to use only 1 thread (e.g. by setting `OMP_NUM_THREADS=1`)

**Comment by drjiadu** [comment]
> We use OpenMP to parallelize random walks and that may be why the results differ. Could you set OpenMP to use only 1 thread (e.g. by setting `OMP_NUM_THREADS=1`)

Yes, I see. Just test with OMP_NUM_THREADS=1. The problem is solved. 
Thanks for your reply.
